sklearn/decomposition/pca.py

```diff
@@ -385,6 +396,7 @@ def _fit_full(self, X, n_components):
         explained_variance_ = (S ** 2) / n_samples
         total_var = explained_variance_.sum()
         explained_variance_ratio_ = explained_variance_ / total_var
+        singular_values_ = S.copy()  # Store the singular values.
```

Comment:
Can't this be calculated by the user as `np.sqrt(explained_variance_ * n_samples)`?